File: Backend/GestionCourriers/src/main/java/esp/irt/courriers/entites/ReceptionCourrier.java
package esp.irt.courriers.entites;


import javax.persistence.*;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;



@Entity
@Data
@AllArgsConstructor
@NoArgsConstructor
public class ReceptionCourrier extends Courrier {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;

    @ManyToOne(cascade = CascadeType.PERSIST) // Many ReceptionCourriers can belong to one Direction
    @JoinColumn(name = "direction")
    private Direction expediteur;

 

    // Les autres attributs et méthodes de l'entité ReceptionCourrier
}

File: Frontend/GestionCourriers/src/app/models/reception-courrier.ts
import { Direction } from './direction';
import { LectureVentilation } from './lecture-ventilation';
import { Courrier } from './courrier';
export class ReceptionCourrier extends Courrier {
  direction?: Direction;
  expediteur?:Direction;
  lectureVentilation?: LectureVentilation;
}
